python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -0,0 +1,750 @@
+		if node is not None :
+			self.__plug = self.__firstValidScenePlug( node )
+			if self.__plug is not None :
+				self.__plugParentChangedConnection = self.__plug.parentChangedSignal().connect(
+					Gaffer.WeakMethod( self.__plugParentChanged ), scoped = True
+				)
+				self.__plugDirtiedConnections.append( node.plugDirtiedSignal().connect( Gaffer.WeakMethod( self.__plugDirtied ), scoped = True ) )
+
+		self.__settingsNode["in"].setInput( self.__plug )
+
+		# call base class update - this will trigger a call to _titleFormat(),
+		# hence the need for already figuring out the plug.
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		self.__setPathListingPath()
+
+	def _updateFromContext( self, modifiedItems ) :
+
+		if any( not i.startswith( "ui:" ) for i in modifiedItems ) :
+			self.__setPathListingPath()
+
+	def _titleFormat( self ) :
+
+		return GafferUI.NodeSetEditor._titleFormat(
+			self,
+			_maxNodes = 1 if self.__plug is not None else 0,
+			_reverseNodes = True,
+			_ellipsis = False
+		)
+
+	@GafferUI.LazyMethod()
+	def __updateColumns( self ) :
+
+		tabGroup = self.__settingsNode["tabGroup"].getValue()
+		currentSection = self.__settingsNode["section"].getValue()
+
+		sectionColumns = []
+
+		for groupKey, sections in self.__columnRegistry.items() :
+			if IECore.StringAlgo.match( tabGroup, groupKey ) :
+				section = sections.get( currentSection or None, {} )
+				sectionColumns += [ c( self.__settingsNode["in"], self.__settingsNode["editScope"] ) for c in section.values() ]
+
+		self.__pathListing.setColumns( [ self.__renderPassNameColumn, self.__renderPassActiveColumn ] + sectionColumns )
+
+	def __settingsPlugSet( self, plug ) :
+
+		if plug in ( self.__settingsNode["section"], self.__settingsNode["tabGroup"] ) :
+			self.__updateColumns()
+
+	def __plugDirtied( self, plug ) :
+
+		if self.__settingsNode["in"] is not None and plug == self.__settingsNode["in"]["globals"] :
+			self.__setPathListingPath()
+
+	def __plugParentChanged( self, plug, oldParent ) :
+
+		# if a plug has been removed or moved to another node, then
+		# we need to stop viewing it - _updateFromSet() will find the
+		# next suitable plug from the current node set.
+		self._updateFromSet()
+
+	@GafferUI.LazyMethod( deferUntilPlaybackStops = True )
+	def __setPathListingPath( self ) :
+
+		if self.__plug is not None :
+			# We take a static copy of our current context for use in the RenderPassPath - this prevents the
+			# PathListing from updating automatically when the original context changes, and allows us to take
+			# control of updates ourselves in _updateFromContext(), using LazyMethod to defer the calls to this
+			# function until we are visible and playback has stopped.
+			contextCopy = Gaffer.Context( self.getContext() )
+			self.__pathListing.setPath( _GafferSceneUI._RenderPassEditor.RenderPassPath( self.__settingsNode["in"], contextCopy, "/", filter = self.__filter ) )
+		else :
+			self.__pathListing.setPath( Gaffer.DictPath( {}, "/" ) )
```

Comment:
When I wrote the ImageInspector, I managed to get rid of this ugly fallback to DictPath. We're making the RenderPassPath with `__settingsNode["in"]`, and there's no harm in doing that with a plug without an input (it's just an empty scene). I _think_ this required some other changes I made on `main` though, because in 1.3 we can't find the ScriptNode (for cancellation support) from the plug when it doesn't have an input.

Could you add a todo to simplify this in 1.4 please, in line with ImageInspector?

Reply:
Todo to simplify in 1.4 added in e2c6238